src/main/scala/HashMaps_Anagrams.scala

```diff
@@ -0,0 +1,74 @@
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+import scala.io.Source
+
+object HashMaps_Anagrams {
+
+  def main(args: Array[String]): Unit = {
+
+
+    add_all(inport(filename))
+    println(most_anagrams())
+  }
+
+  val filename = "WordList.txt"
+  var map = new mutable.HashMap[String, ArrayBuffer[String]]
+
+
+  def most_anagrams(): String = {
+    var max: Int = 0
+    var most: ArrayBuffer[String] = new ArrayBuffer[String]()
+    for (key <- map.keySet) {
+      if (max < map(key).size) {
+        max = map(key).size
+        most = new ArrayBuffer[String]() += key
+
+      }
+      else if (max == map(key).size) most += key
+    }
+    if (most.size == 0) "No anagrams"
```

Comment:
so many if statement... could you not rework the solution a bit to either separate some of them out into helper methods or pattern matching?